servers/rendering/renderer_rd/shader_rd.cpp

```diff
@@ -480,6 +480,7 @@ void ShaderRD::_save_to_cache(Version *p_version, int p_group) {
 }
 
 void ShaderRD::_allocate_placeholders(Version *p_version, int p_group) {
+	ERR_FAIL_COND(p_version->variants == nullptr);
```

Comment:
Prefer `ERR_FAIL_NULL(p_version->variants)` when checking for null pointer.

Reply:
Beat me to it!